lib/extensions.js

```diff
@@ -228,6 +232,11 @@ module.exports = function (realmConstructor) {
                   asyncOpenTask.cancel();
                   asyncOpenTask = null;
                 }
+                // Resolve immediately if the other realm open promise has already finished
+                if (promiseResolved) {
+                  return resolve();
```

Comment:
If anything it should reject, I don't like that it's resolving to `undefined`, if a bug gets introduced in the future this might make `Realm.open` return promises that doesn't resolve to a `Realm`: If anything it should `reject`.

Reply:
I'm happy with rejecting

Reply:
I've updated the code as per your suggestion using `clearTimeout`